### PR TITLE
[APB-1882][AO] Display agent-invitations and tax-history links only for whitelisted agents.

### PR DESCRIPTION
**Pull request by arturopala**

### CODE REVIEW

**Comment by arturopala** [comment]
This PR implements simplified version of APB-1882 where both links on existing agent services account page are displayed depending on agent's whitelisted status. To be confirmed with @nick.bolding

File: app/uk/gov/hmrc/agentservicesaccount/auth/PasscodeVerification.scala [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  def tokenQueryParam[A](implicit request: Request[A]): Option[String] =
+    request.getQueryString(tokenParam).map(token => s"?$tokenParam=$token")
+
+  def throwConfigNotFound(configKey: String) = throw new PasscodeVerificationException(s"The value for the key '$configKey' should be setup in the config file.")
+
+  def addRedirectUrl[A](implicit request: Request[A]): Result => Result = e =>
+    e.addingToSession(SessionKeys.redirect -> buildRedirectUrl(request))
+
+  def buildRedirectUrl[A](req: Request[A]): String =
+    if (env != "Prod") s"http${if (req.secure) "s" else ""}://${req.host}${req.path}" else req.path
```
Comment (opetch) [inline app/uk/gov/hmrc/agentservicesaccount/auth/PasscodeVerification.scala]: No tests assert on the result of this function. Do we need a '/' between host and path?

Comment (opetch) [reply app/uk/gov/hmrc/agentservicesaccount/auth/PasscodeVerification.scala]: thinking about it path will be prefixed with a slash, but we should really have a test checking the session value for both Dev and Prod scenarios

Comment (arturopala) [reply app/uk/gov/hmrc/agentservicesaccount/auth/PasscodeVerification.scala]: good questions, my only excuse is that I copy-pasted it from official HMRC codebase of passcode-verification

**Pull request merged**